### PR TITLE
feat(lib structure and rabbit broker consume): initial lib structure …

### PR DESCRIPTION
**Pull request by Badaueba**
## Problem
Duplicate code to handle communication in event streams between services

## Solution
Nodejs Library in charge of Producing/Consuming stream events 

## Checklist before code review
- [ ] Circle CI checks passed?
- [ ] Your branch is updated(rebased) with master?
- [ ] Did you write tests for your task, including regression tests?

## Reviewer Checklist
- [ ] I checked if the task has the tests
- [ ] I checked if the task has no security breach
- [ ] I checked if the task doesn't have unoptimized queries

## Checklist after merge
- [ ] If it's a _Sentry_ issue, mark it as resolved.
- [ ] Confirm deploy pass
- [ ] Notify product-team

## Deploy Notes
_Notes regarding deployment of the contained body of work. These should note any
new dependencies, new scripts, etc._

**New environment variables**:
- `env var` : env var details

**New scripts**:
- `script` : script details

**New dependencies**:
- `dependency` : dependency details

**New dev dependencies**:
- `dependency` : dependency details

### CODE REVIEW

File: lib/broker/rabbit/queue.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+	private callback(
+		deliveryInfo: DeliveryInfo,
+		properties: any,
+		payload: Record<string, any>,
+		func: Function,
+	) {
+		const eventName = deliveryInfo.routingKey;
+		const event = new Event(eventName, payload);
+		// func(event, context);
```
Comment (m4theuspereira-maker) [inline lib/broker/rabbit/queue.ts:43]: remove useless comment

File: lib/daemon.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Config } from './config';
+
+export class Daemon {
+	config = Config;
+	constructor() {}
+
+	start() {
+		// this.config.broker.getConnection();
```
Comment (m4theuspereira-maker) [inline lib/daemon.ts:8]: this commented function will be used

File: lib/event.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	hasBody() {
+		return this.body !== undefined;
+	}
+
+	hasName() {
+		return this.name !== undefined;
+	}
+
+	private buildPayload() {
+		//nao faz o menor sentido
```
Comment (m4theuspereira-maker) [inline lib/event.ts:36]: missing comment

File: lib/listener.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { Config } from './config';
+import { MissingAttributeError } from './utils/errors';
+
+export class Listener {
+	public static on(eventName: string, callback: Function) {
+		if (eventName.length <= 0) throw new MissingAttributeError('eventName');
```
Comment (m4theuspereira-maker) [inline lib/listener.ts:6]: the right param in error adapter is 'eventName' as string or eventName as it comes from function param?

**Review by m4theuspereira-maker — COMMENTED** [review]
there's an empty index file into lib folder

File: lib/event.ts (hunk 3 above)
Comment (axelzito) [inline lib/event.ts:36]: Why doesn't it make any sense? :thinking:

File: README.md [hunk 5]
```diff
@@ -124,29 +125,30 @@ Config.close_connection();
...
-          `  - Received the "${event.name}" message from ${event.origin}:`
-      );
-      console.log(`     Message: ${event.body}`);
-      console.log("");
-      context.success();
-  });
+	has_events = false;
+
+	await Listener.on('SOME_EVENT', (event: Event, context: Base) => {
+		has_events = true;
```
Comment (willianbvsanches13) [inline README.md:138]: You need to insert .editorconfig file an use 2 spaces to indent code

File: env.example [hunk 6]
```diff
@@ -0,0 +1,4 @@
+RABBIT_URL=amqp://guest:guest@localhost:5672
```
Comment (willianbvsanches13) [inline env.example:1]: You need to remove this file

File: lib/event.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+	origin: string;
+	action: string;
+	destination: string;
+	schemaVersion: number;
+};
+
+export class Event {
+	private headers: EventHeaders;
+	constructor(
+		private name: string,
```
Comment (willianbvsanches13) [inline lib/event.ts:15]: If you don't change the value of private variables you can use "private readonly"  instead of "private"

**Review by willianbvsanches13 — CHANGES_REQUESTED** [review]

File: lib/listener.ts (hunk 4 above)
Comment (Badaueba) [reply lib/listener.ts:6]: should be "Event name" to avoid confusion with param name

File: lib/event.ts (hunk 3 above)
Comment (Badaueba) [reply lib/event.ts:36]: becuase there is a property "body" inside a "body" object, it seems a bit confuse, but honestly i didn't mean to commit with this comment hahahaaha

File: env.example (hunk 6 above)
Comment (Badaueba) [reply env.example:1]: right. there is no need to use docker-compose either hahaha

File: README.md (hunk 5 above)
Comment (Badaueba) [reply README.md:138]: we already have a prettierrc file to handle this, make sure your vscode is properly configured to format code automatically

**Review by m4theuspereira-maker — APPROVED** [review]

**Review by willianbvsanches13 — APPROVED** [review]

**Pull request merged**